tools/makejwt/main.go

```
@@ -0,0 +1,135 @@
+	claims := jwt.MapClaims{
+		"iat": now.Unix(),
+		"exp": now.Add(spec.Duration).Unix(),
+		"iss": spec.AppID,
+	}
+
+	// Create JWT token
+	token := jwt.NewWithClaims(jwt.SigningMethodRS256, claims)
+
+	// Sign the token with the private key
+	tokenString, err := token.SignedString(privateKey)
+	if err != nil {
+		return "", fmt.Errorf("error signing token: %w", err)
+	}
+	if spec.Raw {
+		return tokenString, nil
+	}
+	pubKey, ok := (privateKey.Public()).(*rsa.PublicKey)
+	if !ok {
+		return "", errors.New("error expected RSA public key")
+	}
+	return buildJsonReply(pubKey, tokenString)
+}
+
+func buildJsonReply(publicKey *rsa.PublicKey, jwt string) (string, error) {
+	pubKeyDER, err := asn1.Marshal(*publicKey)
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal public key into ASN1: %w", err)
+	}
+	pubKeyPem := pem.EncodeToMemory(&pem.Block{Type: "RSA PUBLIC KEY", Bytes: pubKeyDER})
+	jsonBytes, err := json.Marshal(jsonReply{PublicKey: string(pubKeyPem), JWT: jwt})
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal public key into PEM: %w", err)
+	}
+	return string(jsonBytes), nil
+}
+
+func printJWT(w io.Writer, args []string) error {
+	var spec JWTSpec
+	fs := flag.NewFlagSet("jwtSpecFlags", flag.ContinueOnError)
+	fs.StringVar(&spec.Base64PEMBytes, "key", "", "Base64 of the PEM key contents")
+	fs.StringVar(&spec.AppID, "appId", "", "Application ID for the JWT token")
+	fs.DurationVar(&spec.Duration, "duration", 10*time.Minute, "Duration the JWT token will be valid for")
+	fs.BoolVar(&spec.Raw, "raw", true, "Emit the raw JWT or a JSON with the jwt and the public key certificate")
+	if err := fs.Parse(args); err != nil {
+		fs.PrintDefaults()
+		return fmt.Errorf("error parsing command line arguments: %w", err)
+	}
+	if spec.Base64PEMBytes == "" {
+		return ErrorEmptyPEMKey
+	}
+	if spec.AppID == "" {
+		return ErrorEmptyAppId
+	}
+	newJWT, err := makeJWT(&spec)
+	if err != nil {
+		log.Fatalf("failed to generate JWT: %v", err)
```

this halts the complete program, this shouldn't happen here, we should bubble up the error and `os.Exit(1)` in main instead.

---

right, will fix!
thx